sources/core/Stride.Core.Mathematics/Matrix.cs

```diff
@@ -453,6 +453,33 @@ public float this[int index]
             }
         }
 
+        /// <summary>
+        /// Casts from System.Numerics to Stride.Maths matrix
+        /// </summary>
+        /// <param name="v">Value to cast</param>
+        public static implicit operator Matrix(System.Numerics.Matrix4x4 v)
```

Comment:
This is in fact a transpose followed by a cast. Better to write it like that, remove the code duplication. Better yet, while you're at it, cast it first, then transpose, so the transpose is optimized. Little effort to then go and replace the `Transpose` method doing it like that, too, with 2 casts - which should both be free.